src/std.rs

```diff
@@ -188,7 +188,7 @@ impl Inner {
         match next {
             None => self.tail = prev,
             Some(n) => unsafe {
-                n.as_ref().next.set(prev);
+                n.as_ref().prev.set(prev);
```

Comment:
what happens here?

Reply:
There was a bug in there that was causing the MIRI issues. It should be fixed now.